### PR TITLE
feat(Inji-402): track login & onboarding flow events in telemetry

### PR DESCRIPTION
**Pull request by PuBHARGAVI**

### CODE REVIEW

File: screens/AuthScreenController.ts [hunk 1]
```diff
@@ -53,10 +66,12 @@ export function useAuthScreen(props: RootRouteProps) {
 
   useEffect(() => {
     if (isAuthorized) {
+      sendEndEvent(getEndEventData('App Onboarding', 'SUCCESS'));
       props.navigation.reset({
         index: 0,
-        routes: [{ name: 'Main' }],
+        routes: [{name: 'Main'}],
       });
+      sendImpressionEvent(getImpressionEventData('App Onboarding', 'Main'));
```
Comment (swatigoel) [inline screens/AuthScreenController.ts:74]: Use "Home Page" instead of "Main"

File: shared/telemetry/TelemetryUtils.js [hunk 2]
```diff
@@ -135,6 +135,33 @@ export function getAppInfoEventData() {
...
+        )
+      : sendErrorEvent(
+          getErrorEventData(
+            'App Login',
+            'mismatch',
+            'Passcode did not match',
+            {},
+          ),
+        );
+    passcodeRetryCount = 1;
```
Comment (swatigoel) [inline shared/telemetry/TelemetryUtils.js:161]: can simplify the logic

File: screens/WelcomeScreenController.ts [hunk 3]
```diff
@@ -55,9 +63,18 @@ export function useWelcomeScreen(props: RootRouteProps) {
...
+        sendStartEvent(getStartEventData('App Login'));
+        sendInteractEvent(
+          getInteractEventData(
+            'App Login',
+            'TOUCH',
+            'Unlock application button',
+          ),
+        );
+        sendImpressionEvent(getImpressionEventData('App Login', 'Passcode'));
+        props.navigation.navigate('Passcode', {setup: isSettingUp});
```
Comment (swatigoel) [inline screens/WelcomeScreenController.ts:77]: Impression Event should be tracked once we are on passcode screen

**Review by swatigoel — CHANGES_REQUESTED** [review]

File: shared/telemetry/TelemetryUtils.js (hunk 2 above)
Comment (PuBHARGAVI) [reply shared/telemetry/TelemetryUtils.js:161]: done

File: screens/PasscodeScreen.tsx [hunk 4]
```diff
@@ -9,16 +9,56 @@ import {PasscodeRouteProps} from '../routes';
...
+            errorId: 'user_cancel',
+            errorMessage: 'Authentication canceled',
+          }),
+        )
+      : sendEndEvent(
+          getEndEventData('App Login', 'FAILURE', {
+            errorId: 'user_cancel',
+            errorMessage: 'Authentication canceled',
+          }),
+        );
```
[the remark this answers is not in the document]
Comment (PuBHARGAVI) [reply screens/PasscodeScreen.tsx:37]: done

File: screens/AuthScreenController.ts (hunk 1 above)
Comment (PuBHARGAVI) [reply screens/AuthScreenController.ts:74]: done

File: screens/WelcomeScreenController.ts (hunk 3 above)
Comment (PuBHARGAVI) [reply screens/WelcomeScreenController.ts:77]: done

**Review by swatigoel — APPROVED** [review]

**Pull request merged**